merge_checks.py

```diff
@@ -28,16 +32,27 @@ def get_base_revision(base_ref: str) -> str:
     return _run_process(f"git rev-parse origin/{base_ref}")
 
 
-def get_subject_markers(head_hash: str, base_hash: str) -> Tuple[str, ...]:
-    log = _run_process(f"git log --pretty='format:%s' {base_hash}..{head_hash}").splitlines()
-    return tuple(line.split(maxsplit=1)[0] for line in log)
+def get_subject(head_hash: str, base_hash: str) -> Sequence[str]:
+    return tuple(_run_process(f"git log --pretty='format:%s' {base_hash}..{head_hash}").splitlines())
+
+
+def get_subject_markers(subjects: Sequence[str]) -> Sequence[str]:
+    return tuple(line.split(maxsplit=1)[0] for line in subjects)
 
 
 def has_merge_commits(head_hash: str, base_hash: str) -> bool:
     parents = _run_process(f"git log --pretty=%p {base_hash}..{head_hash}").splitlines()
     return any(len(line.split(maxsplit=1)) > 1 for line in parents)
 
 
+def has_wrong_commit_message(subject_markers: Sequence[str]) -> Sequence[str]:
+    return tuple(
+        marker
+        for marker in subject_markers
+        if re.match(rf"^({'|'.join(ALLOWED_COMMIT_MESSAGE_TYPES)})((\([a-z\d\s-]+\): .*)|(: .*))", marker) is None
```

Comment:
```suggestion
        if re.match(rf"^({'|'.join(ALLOWED_COMMIT_MESSAGE_TYPES)})\([a-z0-9-]+\): .+", marker) is None
```

Do we really want to allow empty scopes and empty message after scope, right?

Reply:
I checked some repositories and found some message without scope. But for me it is okay to restrict this. Empty messages shouldn't be allowed, I will update the test cases.

Reply:
@zyv Can you explain me why you would prefer `0-9` instead of `\d`?
PyCharm give this hint `'0-9' can be simplified to '\d'`